tests/functional/test_s3.py

```diff
@@ -1340,6 +1341,41 @@ def test_500_error_with_non_xml_body(self):
             self.assertEqual(len(http_stubber.requests), 2)
 
 
+class TestS3Parser(BaseS3OperationTest):
+    def test_valid_expires_value_in_response(self):
+        expires_value = "Thu, 01 Jan 1970 00:00:00 GMT"
+        mock_headers = {'expires': expires_value}
+        s3 = self.session.create_client("s3")
+        with ClientHTTPStubber(s3) as http_stubber:
+            http_stubber.add_response(headers=mock_headers)
+            response = s3.get_object(Bucket='mybucket', Key='mykey')
+            self.assertIn('Expires', response)
+            self.assertEqual(
+                response['Expires'],
+                datetime.datetime(1970, 1, 1, tzinfo=tzutc()),
+            )
+            self.assertIn('ExpiresString', response)
+            self.assertEqual(response['ExpiresString'], expires_value)
+            self.assertEqual(len(http_stubber.requests), 1)
+
+    def test_invalid_expires_value_in_response(self):
+        expires_value = "Invalid Date"
+        mock_headers = {'expires': expires_value}
+        s3 = self.session.create_client("s3")
+        with self.assertLogs('botocore.parsers', level='WARNING') as log:
+            with ClientHTTPStubber(s3) as http_stubber:
+                http_stubber.add_response(headers=mock_headers)
+                response = s3.get_object(Bucket='mybucket', Key='mykey')
+                self.assertNotIn('Expires', response)
+                self.assertIn('ExpiresString', response)
+                self.assertEqual(response['ExpiresString'], expires_value)
+                self.assertIn(
+                    'Failed to parse the "Expires" member as a timestamp',
+                    log.output[0],
+                )
```

Comment:
Can we make this a bit more robust in the event another Warning starts getting raised in this flow? Is it sufficient to check that at least one output in logs contains this warning?